src/parser/ast/mod.rs

```diff
@@ -11,32 +11,81 @@ pub mod tl;
 /// Debug symbol reference, points to the source file, where a AST node comes from.
 #[derive(Clone)]
 pub struct DebugSymRef {
-    /// Starting line number in the file
-    pub line_start: usize,
-    /// Starting column number in the file
-    pub col_start: usize,
-    /// Ending line number in the file
-    pub line_end: usize,
-    /// Ending column number in the file
-    pub col_end: usize,
+    /// Starting byte number in the file
+    pub start: usize,
+    /// Ending byte number in the file
+    pub end: usize,
     /// Source file reference
-    file: Rc<SimpleFile<String, String>>,
+    file: Arc<SimpleFile<String, String>>,
 }
 
 impl DebugSymRef {
-    pub fn new(
-        line_start: usize,
-        col_start: usize,
-        line_end: usize,
-        col_end: usize,
-        file: Rc<SimpleFile<String, String>>,
-    ) -> DebugSymRef {
-        DebugSymRef {
-            line_start,
-            col_start,
-            line_end,
-            col_end,
-            file,
+    pub fn new(start: usize, end: usize, file: Arc<SimpleFile<String, String>>) -> DebugSymRef {
+        DebugSymRef { start, end, file }
+    }
+
+    fn get_column_number(&self, line_index: usize, start: usize) -> usize {
```

Comment:
@alxkzmn why do you think it is better to don't pre-compute the values? For performance?

Reply:
Because the things like rows and columns are file-related so they are needed only in user-facing scenarios like language server.

Reply:
Ok 👍  we will use them in the error messages, but I guess we only need to calculate them for those, so it is good